ddmrp/models/procurement_rule.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp (https://www.camptocamp.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class ProcurementRule(models.Model):
+    _inherit = 'procurement.rule'
+
+    def _prepare_mo_vals(self, product_id, product_qty, product_uom,
+                         location_id, name, origin, values, bom):
+        result = super(ProcurementRule, self)._prepare_mo_vals(
+            product_id, product_qty, product_uom, location_id,
+            name, origin, values, bom
+        )
+        # this field can be passed by
+        # StockWarehouseOrderpoint._prepare_procurement_values
+        # (yes as a recordset!)
+        if values.get('orderpoint_id'):
+            result['orderpoint_id'] = values['orderpoint_id'].id
+        return result
+
+    def _run_manufacture(self, product_id, product_qty, product_uom,
+                         location_id, name, origin, values):
+        super(ProcurementRule, self)._run_manufacture(
+            product_id, product_qty, product_uom,
+            location_id, name, origin, values
+        )
+        orderpoint = values.get('orderpoint_id')
+        if orderpoint:
+            orderpoint.cron_actions()
```

Comment:
This was not done in 9.0 and 10.0. I'm not sure it's good performance-wise. It was necessary for the test `test_buffer_zones_all` to pass after the generation of the `mrp.production` otherwise the stats would not be up-to-date and the recommended quantity would still be 40 instead of 0 (in the last check)

Reply:
if ít's just for tests wouldn't be better to state it w/ a comment and use context key to enable it on demand (and use it tests)

Reply:
I don't think it's just for tests, it was just that I spotted the requirement to call it with an existing test which was failing.

Reply:
I think is good to refresh the buffer/orderpoint after a MO is created, so the MO execution status will be updated too. 

Ideal would be to handle all the updates using the OCA/queue, not in the migration scope though.